src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1252,9 +1252,12 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
                                                        out okToDowngradeToNeither);
                 }
 
+                var type1Normalized = type1.NormalizeTaskTypes(Compilation);
+                var type2Normalized = type2.NormalizeTaskTypes(Compilation);
```

Comment:
Note `NormalizeTaskTypes()` is called in `OverloadResolution.BetterFunctionMember<>` whenever comparing parameter types to ignore differences in `Task` types. And `NormalizeTaskTypes()` walks the type calling `GetAsyncMethodBuilderType()` on each type part, and substituting `System.Threading.Tasks.Task` or `System.Threading.Tasks.Tasks<T>` for any `Task`-like types. However, the overhead should by low since `GetAsyncMethodBuilderType()` is essentially a dictionary lookup, and substitution only occurs if `Task`-like types are found.